type: unknown

Hi there,

looking at the example file `journalbeat.yml` there is no option to add a type, and when loading kibana it shows a: `unkown`
With filebeat that was configured by `document_type`. I placed it everywhere, flushed db, and launched journalbeat but it does not recognize it.

Well, I can't get all this `fields` stuff working at all now.
I receive this:

```sh
$ sudo ./journalbeat
panic: reflect.Set: value of type common.MapStr is not assignable to type *common.MapStr

goroutine 1 [running]:
panic(0x8acd00, 0xc8201b3d20)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
reflect.Value.assignTo(0xa2cfe0, 0xc8201b8f30, 0x15, 0xa7da60, 0xb, 0xa2ac40, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/reflect/value.go:2164 +0x3be
reflect.Value.Set(0xa2ac40, 0xc8200930d8, 0x196, 0xa2cfe0, 0xc8201b8f30, 0x15)
        /usr/local/go/src/reflect/value.go:1334 +0x95
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyGetField(0xc8201b8ab0, 0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0x0, 0x0, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:161 +0x252
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyStruct(0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0x0, 0xa1b7a0, 0xc820093040, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:122 +0xc22
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyMergeValue(0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:298 +0x1026
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyGetField(0xc8201b8150, 0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0x0, 0x0, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:156 +0x1aa
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyStruct(0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0x55d400, 0x98ebc0, 0xc8200a4a00, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:122 +0xc22
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.reifyInto(0xa65ad8, 0x6, 0xa81400, 0x8, 0xa610d8, 0x1, 0x0, 0xa61000, 0x98ebc0, 0xc8200a4a00, ...)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:40 +0x319
github.com/mheese/journalbeat/vendor/github.com/urso/ucfg.(*Config).Unpack(0xc8201b8150, 0x87b240, 0xc8200a4a00, 0xc82014fc90, 0x1, 0x1, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/urso/ucfg/reify.go:24 +0x356
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/common.(*Config).Unpack(0xc8201b8150, 0x87b240, 0xc8200a4a00, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/common/config.go:34 +0xc8
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/cfgfile.Read(0x87b240, 0xc8200a4a00, 0x0, 0x0, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/cfgfile/cfgfile.go:53 +0x88
github.com/mheese/journalbeat/beat.(*Journalbeat).Config(0xc8200a4a00, 0xc82005e900, 0x0, 0x0)
        github.com/mheese/journalbeat/beat/journalbeat.go:75 +0x98
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat.(*instance).config(0xc8200e53e0, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:238 +0x78d
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat.(*instance).launch(0xc8200e53e0, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:296 +0xa0
github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat.Run(0xa7a3d0, 0xb, 0xa611e8, 0x5, 0x7f2df66f5c48, 0xc8200a4a00, 0x0, 0x0)
        github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:126 +0x6d
main.main()
        github.com/mheese/journalbeat/main.go:26 +0x7d
```

Looking into it right now

--- a/beat/config.go
+++ b/beat/config.go
@@ -30,6 +30,7 @@ type JournalReaderConfig struct {
 	MoveMetadataLocation *string        `config:"move_metadata_to_field"`
 	FieldsDest           *string        `config:"fields_dest"`
 	Fields               *common.MapStr `config:"fields"`
+	DefaultType          *string        `config:"default_type"`
 }
 
 // ConfigSettings holds JournalConfig at the Input section of the config file

--- a/beat/journalbeat.go
+++ b/beat/journalbeat.go
@@ -52,6 +52,7 @@ type Journalbeat struct {
 	moveMetadataLocation string
 	fieldsDest           string
 	fields               common.MapStr
+	defaultType          string
 
 	jr   *sdjournal.JournalReader
 	done chan int
@@ -134,6 +135,12 @@ func (jb *Journalbeat) Config(b *beat.Beat) error {
 
 	if jb.JbConfig.Input.Fields != nil {
 		jb.fields = *jb.JbConfig.Input.Fields
+	}
+
+	if jb.JbConfig.Input.DefaultType != nil {
+		jb.defaultType = *jb.JbConfig.Input.DefaultType
+	} else {
+		jb.defaultType = "journal"
 	}
 
 	if _, ok := SeekPositions[jb.seekPosition]; !ok {
@@ -295,7 +302,7 @@ func Publish(beat *beat.Beat, jb *Journalbeat) {
 		// TODO: type should be derived from the system journal
 		_, ok := m["type"].(string)
 		if !ok {
-			m["type"] = "unknown"
+			m["type"] = jb.defaultType
 		}
 
 		// add input_type if it does not exist yet (or if it is not a string)

--- a/etc/journalbeat.yml
+++ b/etc/journalbeat.yml
@@ -18,6 +18,7 @@ input:
   clean_field_names: true
   convert_to_numbers: false
   move_metadata_to_field: journal
+  #default_type: journal
   #fields_dest: my.nested.destination
   #fields:
   #  custom_field: custom_value

